filemanager/controllers/upload.py

```
@@ -93,49 +93,81 @@
+        The unique identifier for the upload_obj in question.
+
+    Returns
+    -------
+    dict
+        Complete summary of upload processing.
+    int
+        An HTTP status code.
+    dict
+        Some extra headers to add to the response.
+
+    """
+
     logger.info(f"{upload_id}: Deleting upload workspace.")
 
-    # TODO: This request was not really part if this sprint. Write down ideas and
-    #      come back to later.
 
     # Need to add several checks here
 
     # At this point I believe we know that caller is authorized to delete the
     # workspace. This is checked at the routes level.
 
-    # Does workspace exist? Has it already been deleted? Generate 400:BadRequest error.
-
+    # Does workspace exist? Has it already been deleted? Generate 400:NotFound error.
     # Do we care is workspace is ACTIVE state? And not released? NO. But log it...
-
+    # Do we want to stash source.log somewhere?
     # Do we care if workspace was modified recently...NO. Log it
 
+
     try:
         # Make sure we have an upload_obj to work with
         upload_obj: Optional[Upload] = uploads.retrieve(upload_id)
 
         if upload_obj is None:
             # Invalid workspace identifier
+            # Note: DB entry will exist for workspace that has already been
+            #       deleted
             raise NotFound(UPLOAD_NOT_FOUND)
         else:
-            # Any other conditions that raise red flags? Any that stop the
-            # deletion?
 
             # Actually remove entire workspace directory structure. Log
-            # something to global log since source log is being removed!
+            # everything to global log since source log is being removed!
 
             # Initiate workspace deletion
 
             # Update database (but keep around) for historical reference. Does not
             # consume very much space. What about source log?
+            # Create Upload object
+            if upload_obj.state == 'DELETED':
```

Might be something for another PR, but consider defining state flags as class attributes. E.g. 

```python
class Upload(Data):
    DELETED = 'DELETED'
```

The advantages of this are that it improves self-documentation of the ``Upload`` class, and it also ensure that you will get feedback if you spell it wrong. E.g.  if ``upload_obj.state == 'DEELTED':`` (note typo) will fail silently, whereas ``upload_obj.state == Upload.DEELTED`` will cause an ``AttributeError``.

---

I've replaced status strings with Upload class attributes. Much cleaner now. Pushed to github.

I will rename other Upload class to UploadWorkspace in a future PR. This will help eliminate any remaining confusion.